staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -1213,11 +1213,12 @@ func expandIfFilePattern(pattern string) ([]string, error) {
 	if _, err := os.Stat(pattern); os.IsNotExist(err) {
 		matches, err := filepath.Glob(pattern)
 		if err == nil && len(matches) == 0 {
-			return nil, fmt.Errorf("no match")
+			return nil, fmt.Errorf("the path %q does not exist", pattern)
 		}
-		if err == nil || err != filepath.ErrBadPattern {
-			return matches, err
+		if err != nil {
+			return nil, fmt.Errorf("pattern %q is not valid: %v", pattern, err)
```

Comment:
this should remain limited to `ErrBadPattern`, right?

Reply:
actually, should be:

```
if err == filepath.ErrBadPattern {
  return nil, fmt.Errorf("pattern %q is not valid: %v", pattern, err)
}
if err != nil {
  return matches, err
}
```

Reply:
Fixed, thx!